lib/bye_bye_bye/mbta_client.ex

```diff
@@ -0,0 +1,135 @@
+defmodule ByeByeBye.MbtaClient do
+  @moduledoc """
+  Client for interacting with the MBTA API.
+  """
+
+  alias ByeByeBye.Config
+  require Logger
+
+  @doc """
+  Gets all current alerts from the MBTA API.
+
+  ## Examples
+
+      iex> MbtaClient.get_alerts()
+      {:ok, alerts}
+      {:error, reason}
+  """
+  def get_alerts do
+    path = "/alerts"
+    do_get_request(path)
+  end
+
+  @doc """
+  Gets schedules filtered by trips or routes and optionally filtered by time window.
+
+  ## Parameters
+    * `opts` - Map or keyword list of options:
+      * `:trips` - Filter by list of trip IDs
+      * `:routes` - Filter by list of route IDs
+      * `:min_time` - Filter schedule after this time (HH:MM format)
+      * `:max_time` - Filter schedule before this time (HH:MM format)
+
+  Only one of trips or routes must be provided, not both.
+
+  ## Examples
+
+      iex> MbtaClient.get_schedules(trips: ["123", "456"])
+      {:ok, schedules}
+
+      iex> MbtaClient.get_schedules(%{routes: ["Red", "Blue"], min_time: "2024-01-20T10:00:00-05:00"})
+      {:ok, schedules}
+  """
+  def get_schedules(opts) when is_map(opts) do
+    opts
+    |> Enum.into([])
+    |> get_schedules()
+  end
+
+  def get_schedules(opts) when is_list(opts) do
+    with {:ok, params} <- validate_schedule_params(opts) do
+      path = "/schedules"
+
+      # Stop sequence is the only field we need to generate correct STUs
+      params = Map.put(params, "fields[schedule]", "stop_sequence")
+
+      case do_get_request(path, params: params) do
+        {:ok, data} -> {:ok, schedules_by_trip(data)}
+        error -> error
+      end
+    end
+  end
+
+  defp validate_schedule_params(opts) do
+    trips = Keyword.get(opts, :trips)
+    routes = Keyword.get(opts, :routes)
+
+    cond do
+      trips && routes ->
+        {:error, "Cannot specify both trips and routes"}
+
+      is_nil(trips) && is_nil(routes) ->
+        {:error, "Must specify either trips or routes"}
+
+      true ->
+        params =
+          opts
+          |> Enum.filter(fn {key, _} -> key in [:min_time, :max_time] end)
```

Comment:
nit:
Could use `Keyword.take/2`
```suggestion
          |> Keyword.take([:min_time, :max_time])
```